lib/haml/railtie.rb

```diff
@@ -1,22 +1,19 @@
-if defined?(ActiveSupport)
-  require 'haml/template/options'
-  ActiveSupport.on_load(:before_initialize) do
-    ActiveSupport.on_load(:action_view) do
-      require "haml/template"
-    end
-  end
-end
+require 'haml/template/options'
 
 module Haml
   class Railtie < ::Rails::Railtie
     initializer :haml do |app|
-      require "haml/template"
-      if defined?(::Sass::Rails::SassTemplate) && app.config.assets.enabled
-        require "haml/sass_rails_filter"
+
+      ActiveSupport.on_load(:action_view) do
```

Comment:
> In any case, the important bit is the on_load, otherwise the bug is not fixed. Currently the initializer runs too early.

If so, I think calling `ActiveSupport.on_load` outside `initializer "haml"` is enough. Why double hooks are required? I don't think recommended by 3rd parti blog is a good reason to do that.

If the intention of `initializer "haml"` is ensuring that `ActiveSupport` is loaded before executing this `ActiveSupport.on_load` and it's really necessary, I think this PR is correct 👍 .